alkira/resource_alkira_connector_akamai_prolexic.go

```diff
@@ -0,0 +1,256 @@
+package alkira
+
+import (
+	"log"
+	"strconv"
+
+	"github.com/alkiranet/alkira-client-go/alkira"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceAlkiraConnectorAkamaiProlexic() *schema.Resource {
+	return &schema.Resource{
+		Description: "Manage Connector for Akamai Prolexic. (BETA)\n\n" +
+			"This resource is still under active development. It may have further " +
+			"changes in the near future. Today, to use this connector, you will need " +
+			"to have onboarded a BYOIP with Do Not Advertise set to `true`. Also, the " +
+			"segment with public IPs needs to be reported to Akamai Representative.",
+		Create: resourceConnectorAkamaiProlexicCreate,
+		Read:   resourceConnectorAkamaiProlexicRead,
+		Update: resourceConnectorAkamaiProlexicUpdate,
+		Delete: resourceConnectorAkamaiProlexicDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "The name of the connector.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"akamai_bgp_asn": {
+				Description: "The Akamai BGP ASN.",
+				Type:        schema.TypeInt,
+				Required:    true,
+			},
+			"akamai_bgp_authentication_key": {
+				Description: "The Akamai BGP Authentication Key.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"byoip_options": &schema.Schema{
+				Description: "BYOIP options.",
+				Type:        schema.TypeSet,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"byoip_prefix_id": {
+							Description: "BYOIP prefix ID.",
+							Type:        schema.TypeInt,
+							Required:    true,
+						},
+						"enable_route_advertisement": {
+							Description: "Whether enabling route advertisement.",
+							Type:        schema.TypeBool,
+							Required:    true,
+						},
+					},
+				},
+				Required: true,
```

Comment:
Stylistically we should put the Required/Optional field above Elem. To be clearer, in the case where Elem is a type or set we should put the required field above Elem. Simply for the sake of readability. Not a deal breaker. Just a thought.